### PR TITLE
Add THPVariable_WrapList helper

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #103822
* #105195
* __->__ #105194

### CODE REVIEW

File: torch/csrc/autograd/python_variable.h [hunk 1]
```diff
@@ -81,3 +81,24 @@ void pushPyOutToStack(
...
+inline PyObject* TPHVariable_WrapList(
+    const torch::autograd::variable_list& inputs) {
+  PyObject* pyinput = PyList_New(inputs.size());
+  for (const auto i : c10::irange(inputs.size())) {
+    PyList_SET_ITEM(pyinput, i, THPVariable_Wrap(inputs[i]));
+  }
+  return pyinput;
+}
+
+inline torch::autograd::variable_list TPHVariable_UnpackList(
```
Comment (soulitzer) [inline torch/csrc/autograd/python_variable.h:94]: nit: TPHVariable -> THPVariable

**Review by soulitzer — APPROVED** [review]
Thanks!
The asymmetry of the names is unfortunate, but looks like its preexisting

**Review by albanD — APPROVED** [review]
SGTM

**Comment by jansel** [comment]
> Thanks! The asymmetry of the names is unfortunate, but looks like its preexisting

Yeah, I was trying to match the style of the file.

**Pull request closed without merging**